--- aidsen.hpp	2021-03-08 20:42:04.342630541 +0100
+++ aidsen.hpp	2021-03-08 20:42:00.677453079 +0100
@@ -125,8 +125,8 @@
 //   Jarosław Wiosna (github:JaroslawWiosna)
 //   Danil Kolumbet (github:kolumb)
 
-#ifndef AIDS_HPP_
-#define AIDS_HPP_
+#ifndef AIDSEN_HPP_
+#define AIDSEN_HPP_
 
 #include <cassert>
 #include <cctype>
@@ -136,7 +136,7 @@
 #include <cstdlib>
 #include <cstring>
 
-namespace aids
+namespace aidsen
 {
     ////////////////////////////////////////////////////////////
     // ALLOCATORS
@@ -278,7 +278,7 @@
 #define DEFER_1(x, y) x##y
 #define DEFER_2(x, y) DEFER_1(x, y)
 #define DEFER_3(x)    DEFER_2(x, __COUNTER__)
-#define defer(code)   auto DEFER_3(_defer_) = ::aids::defer_func([&](){code;})
+#define defer(code)   auto DEFER_3(_defer_) = ::aidsen::defer_func([&](){code;})
 
     ////////////////////////////////////////////////////////////
     // MAYBE
@@ -1272,4 +1272,4 @@
     }
 }
 
-#endif  // AIDS_HPP_
+#endif  // AIDSEN_HPP_
