--- aidsen.hpp	2021-03-08 21:52:51.241822363 +0100
+++ aidsen.hpp	2021-03-08 21:52:47.332738019 +0100
@@ -1270,6 +1270,40 @@
             mtor.dealloc(hash_map.buckets, hash_map.capacity);
         }
     }
+
+    ////////////////////////////////////////////////////////////
+    // Stack
+    ////////////////////////////////////////////////////////////
+
+    template <typename T, typename Ator = Mtor>
+    struct Stack
+    {
+        Dynamic_Array<T, Ator> da{};
+
+        void push(T item) {
+            da.push(item);
+        }
+
+        T top() {
+            return da.data[da.size - 1];
+        }
+
+        T pop() {
+            auto ret = top();
+            da.size--;
+            return ret;
+        }
+
+        bool empty() const {
+            return (0 == da.size);
+        }
+    };
+
+    template <typename T>
+    void destroy(Stack<T> stack)
+    {
+        destroy(stack.da);
+    }
 }
 
 #endif  // AIDSEN_HPP_
